epistemix_platform/src/epistemix_platform/services/results_packager.py

```
@@ -0,0 +1,238 @@
+        # Step 1: Validate directory exists
+        self._validate_directory_exists(results_dir)
+
+        # Step 2: Find RUN* directories
+        run_dirs = self._find_run_directories(results_dir)
+        is_single_run_dir = self._is_run_directory(results_dir)
+
+        if not run_dirs and not is_single_run_dir:
+            raise InvalidResultsDirectoryError(
+                f"No FRED output directories (RUN*) found in {results_dir}. "
+                "Expected either a RUN* directory or a parent containing RUN*/ subdirectories."
+            )
```

Ruff (0.14.1)

Avoid specifying long messages outside the exception class

(TRY003)